### PR TITLE
Add sweat.claim load in Locust

### PR DESCRIPTION
**Pull request by akashin**
This PR introduces `sweat.claim` contract necessary for https://github.com/near/nearcore/issues/11107 and uses it to implement `sweat::defer_batch` and `sweat.claim::get_claimable_balance_for_account` calls.

I originally intended to use `sweat.claim::claim` but it requires the balance to be locked for some time to be claimed, so to avoid changing `sweat.claim` contract, I've used `get_claimable_balance_for_account` method instead, which does a similar amount of work and can be called unconditionally.

You can run this with:

```sh
nearup run localnet --binary-path ../nearcore/target/release/ --num-nodes 1 --num-shards 1 --override
export KEY=~/.near/localnet/node0/validator_key.json
locust -H 127.0.0.1:3030 \
  -f locustfiles/sweat.py \
  --funding-key=$KEY --headless -u 10
```

### CODE REVIEW

File: pytest/tests/loadtest/locust/locustfiles/sweat.py [hunk 1]
```diff
@@ -94,10 +101,39 @@ def on_start(self):
...
+        logger.debug(
+            f"{self.account_id} ready to use Sweat contract {self.sweat.account.key.account_id}"
+        )
+
+
+class SweatUser(NearUser):
+    """
+    Registers itself on an FT contract in the setup phase, then does one of two things:
+      - Sends Sweat to random users.
+      - Checks its' balance in sweat.claim contract.
```
Comment (nagisa) [inline pytest/tests/loadtest/locust/locustfiles/sweat.py:116]: But it does not attempt to claim anything, is that right?

**Review by nagisa — APPROVED** [review]

File: pytest/tests/loadtest/locust/locustfiles/sweat.py (hunk 1 above)
Comment (nagisa) [reply pytest/tests/loadtest/locust/locustfiles/sweat.py:116]: Aha, never mind, saw the PR description.

**Review by mooori — APPROVED** [review]
The cost of executing `claim` resp. `get_claimable_balance_for_account` depends on state: it increases with the number of accruals stored for `account_id`. I'm still wrapping my head around nearcore's Locust setup for Sweat, but it seems like running a given `locust` command twice can result in different numbers of storage accesses as the accounts which receive accruals are chosen randomly. This could make it hard to assess the performance impact of nearcore changes based on this workload.

If this interpretation is correct, should we make the generation of accruals deterministic? For instance by introducing a parameter to set the number of accruals per account. That could be done in a follow-up PR.

**Comment by akashin** [comment]
> The cost of executing `claim` resp. `get_claimable_balance_for_account` depends on state: it increases with the number of accruals stored for `account_id`. I'm still wrapping my head around nearcore's Locust setup for Sweat, but it seems like running a given `locust` command twice can result in different numbers of storage accesses as the accounts which receive accruals are chosen randomly. This could make it hard to assess the performance impact of nearcore changes based on this workload.
> 
> If this interpretation is correct, should we make the generation of accruals deterministic? For instance by introducing a parameter to set the number of accruals per account. That could be done in a follow-up PR.

Yes, I think that is a very good observation. Here I'm just doing plumbing to enable calling these smart contract methods. We need to do another step where we tune the Locust workload to actually exercise the network in interesting ways - what you suggest makes a lot of sense.

File: pytest/tests/loadtest/locust/README.md [hunk 2]
```diff
@@ -129,6 +129,7 @@ Currently supported load types:
 | Congestion | congestion.py | (`--congestion-wasm $WASM_PATH`) | Creates a single instance of Congestion contract. Users run large and long transactions. |
 | Sweat (normal load) | sweat.py | (`--sweat-wasm $WASM_PATH`) | Creates a single instance of the SWEAT contract. A mix of FT transfers and batch minting with batch sizes comparable to mainnet observations in summer 2023. |
 | Sweat (storage stress test) | sweat.py | `--tags=storage-stress-test` <br> (`--sweat-wasm $WASM_PATH`) | Creates a single instance of the SWEAT contract. Sends maximally large batches to mint more tokens, thereby touching many storage nodes per receipt. This load will take a while to initialize enough Sweat users on chain. |
+| Sweat (claim) | sweat.py | `--tags=claim-test` <br> (`--sweat-wasm $WASM_PATH`) | Creates a single instance of the SWEAT and SWEAT.CLAIM contract. Sends deferred batches to mint more tokens, thereby touching many storage nodes per receipt. Then calls balance checks that iterate through populated state. |
```
[the remark this answers is not in the document]
Comment (akashin) [reply pytest/tests/loadtest/locust/README.md]: Added.

**Pull request merged**